src/widgets/WordListWidget.vue

```diff
@@ -0,0 +1,87 @@
+<template>
+  <div class="word-list-widget u-widget u-flex">
+    <WordList v-if="filteredWordList.length > 0" :wordList="filteredWordList" />
+  </div>
+</template>
+
+<script>
+  import WIDGETS_NS, { WordList } from '@scaife-viewer/scaife-widgets';
+  import { MODULE_NS } from '@/reader/constants';
+
+  export default {
+    name: 'WordListWidget',
+    scaifeConfig: {
+      displayName: 'Word List',
+    },
+    components: {
+      WordList,
+    },
+    created() {
+      if (this.enabled) {
+        this.fetchData();
+      }
+    },
+    data() {
+      return {
+        wordList: [],
+      };
+    },
+    watch: {
+      passage: 'fetchData',
+    },
+    computed: {
+      selectedToken() {
+        return this.$store.state[MODULE_NS].selectedToken;
+      },
+      filteredWordList() {
+        if (this.selectedToken === null) {
+          return this.wordList;
+        }
+        return this.wordList.filter(w => w.text === this.selectedToken.value);
+      },
+      enabled() {
+        // TODO: Stubbed until metadata ingestion update.
+        // return this.passage && this.metadata.lang === 'grc';
+        return this.passage;
+      },
+      metadata() {
+        return this.$store.getters[`${WIDGETS_NS}/metadata`];
+      },
+      passage() {
+        return this.$store.getters[`${WIDGETS_NS}/passage`];
+      },
+      endpoint() {
+        return 'https://vocab.perseus.org/word-list';
+      },
+      url() {
+        const params = 'page=all&amp;o=1';
+        return this.passage
+          ? `${this.endpoint}/${this.passage}/json/?${params}`
+          : null;
+      },
+    },
+    methods: {
+      fetchData() {
```

Comment:
there is also a fallback we have on the scaife-viewer/scaife-viewer widget (if the word list does not have the particular passage, we query for the lemmas directly)

https://github.com/scaife-viewer/scaife-viewer/blob/227a933c2b803bdf9d247f2d201219f07301e209/static/src/js/reader/widgets/WidgetWordList.vue#L58

Reply:
I'm not sure how this fallback should work.  is the `selectedLemmas` in our case here just `[this.selectedToken.lemma]`?

Reply:
I'm clicking around a lot and not getting any 404s

Reply:
@paltman vocab.perseus.org was something @jtauber cooked up to support the PDL.  You would see 404s for any work not in the PDL, such as the Venetus A edition of the Iliad:

https://deploy-preview-30--explorehomer-dev.netlify.app/reader?urn=urn%3Acts%3AgreekLit%3Atlg0012.tlg001.msA%3A1.1-1.25

We haven't currently lemmatized the Venetus A tokens, so there actually wouldn't be anything to pass to the fallback here.

I think the "fallback" functionality is something we want in the word list widget, but I think we should split it off to a future story because we're blocked on data within EH at the moment.

![](https://github.trello.services/images/mini-trello-icon.png) [Backport fallback from Word List widget](https://trello.com/c/5iDnacZK)